### PR TITLE
add __Init__ to inference.

### PR DESCRIPTION
**Pull request by AnandInguva**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue, if applicable. This will automatically link the pull request to the issue.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/#make-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by AnandInguva** [comment]
Run PythonDocs PreCommit

**Comment by AnandInguva** [comment]
R: @tvalentyn

File: sdks/python/apache_beam/ml/inference/pytorch.py [hunk 1]
```diff
@@ -75,10 +75,13 @@ def __init__(
...
-    state_dict_path: path to the saved dictionary of the model state.
-    model_class: class of the Pytorch model that defines the model structure.
-    device: the device on which you wish to run the model. If ``device = GPU``
-        then device will be cuda if it is available. Otherwise, it will be cpu.
+    Initializes a PytorchModelLoader
+    :param state_dict_path: path to the saved dictionary of the model state.
+    :param model_class: class of the Pytorch model that defines the model
+    structure.
+    :param device: the device on which you wish to run the model. If
+    ``device = GPU`` then device will be cuda if it is avaiable. Otherwise,
```
Comment (tvalentyn) [inline sdks/python/apache_beam/ml/inference/pytorch.py:83]:
```suggestion
    ``device = GPU`` then a GPU device will be used if it is available. Otherwise,
```

File: sdks/python/tox.ini [hunk 2]
```diff
@@ -144,6 +144,7 @@ deps =
   sphinx_rtd_theme==0.4.3
   docutils<0.18
   Jinja2==3.0.3 # TODO(BEAM-14172): Sphinx version is too old.
+  torch==1.9.0
```
Comment (tvalentyn) [inline sdks/python/tox.ini:147]: I don't think we should hardcode a particular version of torch, since Beam doesn't have a requirement on torch.

File: sdks/python/apache_beam/ml/inference/pytorch.py [hunk 3]
```diff
@@ -75,10 +75,13 @@ def __init__(
...
+    it will be cpu.
```
Comment (tvalentyn) [inline sdks/python/apache_beam/ml/inference/pytorch.py:84]:
```suggestion
    it will be CPU.
```

File: sdks/python/apache_beam/ml/inference/api.py [hunk 4]
```diff
@@ -14,6 +14,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
+# mypy: ignore-errors
```
Comment (tvalentyn) [inline sdks/python/apache_beam/ml/inference/api.py:17]: Why is this necessary now?

**Comment by tvalentyn** [comment]
also R: @yeandy

File: sdks/python/apache_beam/ml/inference/api.py (hunk 4 above)
Comment (AnandInguva) [reply sdks/python/apache_beam/ml/inference/api.py:17]: TypeVar should have a bounded source. Right now, we want to keep it generic as possible. So, only this file will be ignored for mypy for now and will circle back to this later. 

Jira related to this: https://issues.apache.org/jira/browse/BEAM-14217

**Comment by AnandInguva** [comment]
PTAL @tvalentyn

**Review by yeandy — APPROVED** [review]
lgtm

**Comment by yeandy** [comment]
Run Python PreCommit

File: sdks/python/apache_beam/ml/inference/api.py (hunk 4 above)
Comment (tvalentyn) [reply sdks/python/apache_beam/ml/inference/api.py:17]: Please add relevant comments to BEAM-14217 and consider adding a # TODO (BEAM-14217): ... 
here if applicable at a later change.
